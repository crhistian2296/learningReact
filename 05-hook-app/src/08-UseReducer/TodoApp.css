p {
  margin: 0;
  padding: 0;
}

.complete {
  text-decoration: line-through;
}
