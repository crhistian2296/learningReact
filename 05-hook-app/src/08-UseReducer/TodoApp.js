import React, { useEffect, useReducer } from 'react';
import TodoAdd from './TodoAdd';
import './TodoApp.css';
import TodoList from './TodoList';
import { todoReducer } from './todoReducer';

// const initialState = [
//   {
//     id: Date.now(),
//     description: 'Aprender React',
//     done: false,
//   },
// ];

const init = () => JSON.parse(localStorage.getItem('todos')) || [];

const TodoApp = () => {
  const [todosList, dispatch] = useReducer(todoReducer, [], init);

  useEffect(() => {
    localStorage.setItem('todos', JSON.stringify(todosList));
    // console.log(todosList);
  }, [todosList]);

  const handleAdd = (newTodo) => {
    dispatch({
      type: 'add',
      payload: newTodo,
    });
  };

  const handleDelete = (todoId) => {
    dispatch({
      type: 'delete',
      payload: todoId,
    });
  };

  const handleToggle = (todoId) => {
    dispatch({
      type: 'toggle',
      payload: todoId,
    });
  };

  return (
    <div className='m-5'>
      <h1>TodoApp: {todosList.length}</h1>
      <hr />
      <div className='row'>
        {/* <div className='col-sm-5'> */}
        <TodoList todosList={todosList} handleDelete={handleDelete} handleToggle={handleToggle} />
        {/* </div> */}
        <div className='col-1'></div>
        <TodoAdd handleAdd={handleAdd} />
      </div>
    </div>
  );
};

export default TodoApp;
