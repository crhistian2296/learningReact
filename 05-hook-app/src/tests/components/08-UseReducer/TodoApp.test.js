import { act } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import React from 'react';
import TodoApp from '../../../08-UseReducer/TodoApp';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas en <TodoApp/>', () => {
  const wrapper = shallow(<TodoApp />);
  Storage.prototype.setItem = jest.fn(() => {});
  test('Debe de renderizar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de agregar un TODO', () => {
    const wrapper2 = mount(<TodoApp />);

    act(() => {
      wrapper2.find('TodoAdd').prop('handleAdd')(demoTodos[0]);
      wrapper2.find('TodoAdd').prop('handleAdd')(demoTodos[1]);
    });

    expect(wrapper2.find('h1').text().trim()).toBe('TodoApp: 2');
    expect(localStorage.setItem).toHaveBeenCalledTimes(2);
  });

  test('Debe de borrar un TODO', () => {
    wrapper.find('TodoAdd').prop('handleAdd')(demoTodos[0]);
    wrapper.find('TodoList').prop('handleDelete')(demoTodos[0].id);

    expect(wrapper.find('h1').text().trim()).toBe('TodoApp: 0');
  });
});
