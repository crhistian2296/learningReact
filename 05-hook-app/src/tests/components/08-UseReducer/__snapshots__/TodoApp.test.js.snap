// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoApp/> Debe de renderizar correctamente 1`] = `
<div
  className="m-5"
>
  <h1>
    TodoApp: 
    0
  </h1>
  <hr />
  <div
    className="row"
  >
    <TodoList
      handleDelete={[Function]}
      handleToggle={[Function]}
      todosList={Array []}
    />
    <div
      className="col-1"
    />
    <TodoAdd
      handleAdd={[Function]}
    />
  </div>
</div>
`;
